pennylane/_qubit_device.py

```diff
@@ -716,10 +721,26 @@ def statistics(self, circuit: QuantumScript, shot_range=None, bin_size=None):
             * Finally, we repeat the measurement statistics for the final 100 shots,
               ``shot_range=[35, 135]``, ``bin_size=100``.
         """
-        observables = circuit.observables
+        if isinstance(circuit, QuantumScript):
+            measurements = circuit.measurements
+
+        else:
+            warnings.warn(
+                message="The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. "
+                "Please use ``circuit`` instead.",
+                category=UserWarning,
+            )
+            measurements = circuit if circuit is not None else observables
+
         results = []
 
-        for obs in observables:
+        for m in measurements:
+            if isinstance(m, MeasurementProcess) and m.obs is not None:
+                obs = m.obs
+                obs.return_type = m.return_type
+            else:
+                obs = m
```

Comment:
Is this only for backward compatibility? In the new system can we pass the measurement process directly in all cases? If so, let's add a TODO here to remove this.

Reply:
Done!